### PR TITLE
Cleanup and reconciliation attempt

### PR DESCRIPTION
**Pull request by m-reza-rahman**

### CODE REVIEW

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import java.nio.file.Files
+import org.apache.commons.io.FileUtils
+
+def jakartaVersion = request.properties["jakartaVersion"].trim()
+def docker = request.properties["docker"].trim()
+def outputDirectory = new File(request.getOutputDirectory(), request.getArtifactId())
+
+validateInput(jakartaVersion, outputDirectory);
+
+switch (request.properties["runtime"]) {
```
Comment (OndroMih) [inline archetype/src/main/resources/META-INF/archetype-post-generate.groovy:10]: I recommend adding a bit more structure to the code. This switch block could be moved to a separate method, e.g. `generateRuntime()`

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            
+        break
+
+    default: println "No runtime will be included in the sample"
+}
+
+// Jakarta version specific processing
+bindEEPackage(jakartaVersion, outputDirectory)
+
+// Remove Dockerfile if not requested or possible
```
Comment (OndroMih) [inline archetype/src/main/resources/META-INF/archetype-post-generate.groovy:36]: This block could go to a method like `generateDocker()`

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+if (docker.equalsIgnoreCase("no")) {
+    println "Docker support was not requested"
+    FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))
+} else if (request.properties["runtime"].equalsIgnoreCase("none")) {
+    println "WARNING: Docker support is not possible without choosing a runtime"
+    FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))    
+}
+
+// Adding Maven Wrapper
+println "Adding Maven Wrapper"
```
Comment (OndroMih) [inline archetype/src/main/resources/META-INF/archetype-post-generate.groovy:46]: Code to generate the Maven wrapper could go to a separate method, like `generateMavenWrapper()`

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 1 above)
Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy]: Good point. Done!

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 2 above)
Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy]: Good point. Done!

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 3 above)
Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy]: Good point. Done!

**Comment by m-reza-rahman** [comment]
I will add support for both WildFly and Open Liberty in the next few days. Things usually slow down for me a bit during weekdays. In the meantime, it’s fine to merge this if there is general agreement on the way forward together.

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, GlassFish does not support the Core Profile")
+    }
+
+    if (runtime.equalsIgnoreCase("glassfish") && (jakartaVersion != '8') && (javaVersion == '8')) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, GlassFish 7 does not support Java SE 8")
+    }
+
+    if (runtime.equalsIgnoreCase("tomee") && (jakartaVersion == '10')) {
```
Comment (IvoNet) [inline archetype/src/main/resources/META-INF/archetype-post-generate.groovy:39]: There are three "if tomee" parts between line 39 and 44. is a sub if not more reasonable here?

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            
+            break
+
+        default: println "No runtime will be included in the sample"
+    }
+}
+
+private bindEEPackage(jakartaVersion, File outputDirectory) {
+    def eePackage = 'javax';
+    if (jakartaVersion != '8') {
```
Comment (IvoNet) [inline archetype/src/main/resources/META-INF/archetype-post-generate.groovy:86]: I would turn this around. Asking positive questions is easier to understand for the reader.

Make the default jakarta ee and only change to javax when `jakartaVersion == 8`

File: archetype/pom.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	<name>Eclipse Starter for Jakarta EE</name>
+	<description>
+		This is the official Eclipse Foundation starter for Jakarta EE.
+		It generates code to help get started with Jakarta EE projects. 
+		It is possible to do so using Maven archetypes. 
+  </description>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (IvoNet) [inline archetype/pom.xml:21]: Does this project really need to run on java 8? we are at java 19 already and 8 has been deprecated for a while now.

File: archetype/src/main/resources/archetype-resources/pom.xml [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+		#end
+	</dependencies>
+
+	<build>
+		<finalName>jakartaee-hello-world</finalName>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.8.0</version>
```
Comment (IvoNet) [inline archetype/src/main/resources/archetype-resources/pom.xml:86]: Lets put all versions in the <properties> section for easy of maintenance

**Review by IvoNet — COMMENTED** [review]
Have made some comments but they are not blocking as far as I am concerned

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 4 above)
Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy:39]: Do you mean a nested if? If so, yes, easy to do.

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 5 above)
Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy:86]: Sure, will change it ASAP.

File: archetype/pom.xml (hunk 6 above)
Comment (m-reza-rahman) [reply archetype/pom.xml:21]: I believe we have general agreement on this already. We need to support Jakarta EE 8 and the required baseline for Jakarta EE 8 is Java SE 8. Moreover, in our customer base, Java SE 8 and Jakarta EE 8 remain extremely popular. You can take a look at major reports like this one and see how popular Java SE 8 still is: https://newrelic.com/resources/report/2022-state-of-java-ecosystem.

If we feel this needs more discussion, let's kind start and thread on the mailing list?

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 4 above)
Comment (IvoNet) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy:39]: yes nested :-) makes it easier to understand

Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy:39]: OK, will address it ASAP.

Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy]: It's addressed. Feel free to mark resolved?

File: archetype/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 5 above)
Comment (m-reza-rahman) [reply archetype/src/main/resources/META-INF/archetype-post-generate.groovy]: It's addressed. Feel free to mark resolved?

File: archetype/src/main/resources/archetype-resources/pom.xml (hunk 7 above)
Comment (m-reza-rahman) [reply archetype/src/main/resources/archetype-resources/pom.xml]: It's addressed. Feel free to mark resolved?

**Review by OndroMih — APPROVED** [review]

**Comment by m-reza-rahman** [comment]
WildFly support added. Open Liberty is next.

**Comment by m-reza-rahman** [comment]
Liberty support done.

**Pull request merged**